Document Console endpoints (#361)
discord-userdoccers/discord-userdoccers: pages/resources/connections.mdx

> # Connections

Reviewer: ```suggestion
# Connected Accounts
```
clearer

> Connections are links of third party accounts to Discord account.

Reviewer: ```suggestion
Connections are links between third party accounts to Discord accounts.
```

| ----- | -------- | ------------------------------------------------ |
| 0     | NONE     | Invisible to everyone except the user themselves |
| 1     | EVERYONE | Visible to everyone                              |

[Partial connections](#partial-connection-structure) always have a visibility of 1.

### Console Device Object

###### Console Device Structure

| Field    | Type      | Description                                                                                     |
| -------- | --------- | ----------------------------------------------------------------------------------------------- |
| id       | snowflake | The ID of the device                                                                            |
| name     | string    | The name of the device                                                                          |
| platform | string    | The [console platform](#connection-type) (only `playstation` and `playstation-stg` are allowed) |

###### Console Handoff Type

| Value                  | Description                                 |
| ---------------------- | ------------------------------------------- |

> | CREATE_NEW_CALL        | Create a new call on console device         |

Reviewer: ```suggestion
| CREATE_NEW_CALL        | Create a new call on a console device         |
```

> | TRANSFER_EXISTING_CALL | Transfer an existing call to console device |

Reviewer: ```suggestion
| TRANSFER_EXISTING_CALL | Transfer an existing call to a console device |
```

| metadata_visibility | integer | [Visibility](#visibility-type) of the connection's metadata             |
| visibility          | integer | [Visibility](#visibility-type) of the connection                        |

<RouteHeader method="DELETE" url="/users/@me/connections/{connection.type}/{connection.id}">
  Delete User Connection
</RouteHeader>

Deletes a connection. Returns a 204 empty response on success. Fires a [User Connections Update](/topics/gateway-events#user-connections-update) and optionally a [Guild Delete](/topics/gateway-events#guild-delete) Gateway event.

<Alert type="warn">

Deleting a connection will remove you from any guilds you joined via the connection's [integrations](#connection-integration-structure).

</Alert>

<RouteHeader method="GET" url="/users/@me/linked-connections" supportsOAuth2="connections">
  Get User Linked Connections
</RouteHeader>

<Alert type="warn">

This endpoint is only usable with an OAuth2 access token with the `connections` scope.

</Alert>

Returns a list of [connection](#connection-object) objects that have a two-way link with the application making the request.

> <RouteHeader method="POST" url="/consoles/connect-request">

Reviewer: oauth2 scope?

>   Get Connect Nonce

Reviewer: ```suggestion
  Create Console Connection
```

</RouteHeader>

> Returns a nonce for connecting to voice on PlayStation console.

Reviewer: ```suggestion
Returns a nonce for connecting to voice on PlayStation consoles.
```

###### JSON Params

| Field                | Type                                                                       | Description                       |
| -------------------- | -------------------------------------------------------------------------- | --------------------------------- |
| analytics_properties | [connect request properties](#connect-request-properties-structure) object | The properties used for analytics |

###### Connect Request Properties Structure

| Field        | Type   | Description                                       |
| ------------ | ------ | ------------------------------------------------- |

> | handoff_type | string | The [console handoff type](#console-handoff-type) |

Reviewer: this enum should be under here

###### Response Body

| Field | Type   | Description |
| ----- | ------ | ----------- |
| nonce | string | The nonce   |

###### Example Response

```json
{ "nonce": "fgnmC0tT" }
```

<RouteHeader method="DELETE" url="/consoles/connect-request/{nonce}">

>   Cancel Connect Request

Reviewer: ```suggestion
  Cancel Console Connection Request
```

</RouteHeader>

Cancels a connect request. Returns a 204 empty response on success.

<RouteHeader method="GET" url="/consoles/{connection.type}/devices">
  Get Console Devices
</RouteHeader>

Returns a list of [console device](#console-device-object) objects.

###### Example Response

```json
[
  {
    "id": "1371598138300956672",
    "name": "My Gifted PlayStation 5",
    "platform": "playstation"
  }
]
```

<RouteHeader method="POST" url="/consoles/{connection.type}/devices/{device.id}/commands">

>   Send Connect Voice Command

Reviewer: ```suggestion
  Send Console Command
```

</RouteHeader>

Sends a command to connect to a voice call on console device.

###### JSON Params

| Field      | Type      | Description                                                              |
| ---------- | --------- | ------------------------------------------------------------------------ |
| command    | string    | The [command type](#console-command-type)                                |
| channel_id | snowflake | The ID of the channel to connect to                                      |
| guild_id?  | snowflake | The ID of the guild the channel is in                                    |
| nonce?     | string    | The nonce obtained from [Get Connect Nonce](#get-connect-nonce) endpoint |

###### Response Body

| Field | Type      | Description                |
| ----- | --------- | -------------------------- |
| id    | snowflake | The ID of the sent command |

<RouteHeader method="DELETE" url="/consoles/{connection.type}/devices/{device.id}/commands/{command.id}">

>   Cancel Command

Reviewer: ```suggestion
  Cancel Console Command
```
